enterprise/micronaut/src/org/netbeans/modules/micronaut/db/MicronautEndpointRequestProvider.java

```diff
@@ -0,0 +1,209 @@
+                                            }
+                                        }
+                                        id2params.put(id, param);
+                                    }
+                                }
+                                if (id2params.isEmpty()) {
+                                    future.complete(null);
+                                } else if (id2params.size() == 1 && id2params.keySet().iterator().next().isEmpty()) {
+                                    VariableElement param = id2params.values().iterator().next();
+                                    StringBuilder sb = new StringBuilder();
+                                    if (fill(cc, 0, null, param, sb)) {
+                                        future.complete(sb.toString());
+                                    } else {
+                                        future.complete("// TODO: Fill the value for parameter '" + param.getSimpleName() + "' of type '" + Utils.getTypeName(cc, param.asType(), false, false) + "'");
+                                    }
+                                } else {
+                                    StringBuilder sb = new StringBuilder().append("{\n");
+                                    for (Iterator<Map.Entry<String, VariableElement>> it = id2params.entrySet().iterator(); it.hasNext();) {
+                                        Map.Entry<String, VariableElement> entry = it.next();
+                                        if (fill(cc, 1, entry.getKey(), entry.getValue(), sb)) {
+                                            if (it.hasNext()) {
+                                                sb.append(',');
+                                            }
+                                        } else {
+                                            sb.append("// TODO: Fill the value for parameter '").append(entry.getKey()).append("' of type '").append(Utils.getTypeName(cc, entry.getValue().asType(), false, false)).append("'");
+                                        }
+                                        sb.append('\n');
+                                    }
+                                    sb.append('}');
+                                    future.complete(sb.toString());
+                                }
+                            } else {
+                                future.complete(null);
+                            }
+                        }
+                    }, true);
+                }
+            }
+        } catch (IOException ex) {
+            future.completeExceptionally(ex);
+        }
+        return future;
+    }
+
+    private static boolean fill(CompilationInfo info, int level, String name, VariableElement ve, StringBuilder sb) {
+        if (name != null) {
+            for (int i = 0; i < level; i++) {
+                sb.append("  ");
+            }
+            sb.append('"').append(name).append("\":");
+            return fill(info, level, null, ve, sb);
```

Comment:
Q: is the return here necessary -- it seems that none of the input vars are changed, `null` name only skips this first name != null part. Maybe just continue ?